uniprot-search/src/test/java/org/uniprot/store/search/document/taxonomy/TaxonomyDocumentConverterTest.java

```diff
@@ -0,0 +1,139 @@
+package org.uniprot.store.search.document.taxonomy;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+import org.uniprot.core.json.parser.taxonomy.TaxonomyJsonConfig;
+import org.uniprot.core.taxonomy.TaxonomyEntry;
+import org.uniprot.core.taxonomy.TaxonomyLineage;
+import org.uniprot.core.taxonomy.TaxonomyRank;
+import org.uniprot.core.taxonomy.TaxonomyStatistics;
+import org.uniprot.core.taxonomy.impl.TaxonomyEntryBuilder;
+import org.uniprot.core.taxonomy.impl.TaxonomyLineageBuilder;
+import org.uniprot.core.taxonomy.impl.TaxonomyStatisticsBuilder;
+import org.uniprot.core.taxonomy.impl.TaxonomyStrainBuilder;
+import org.uniprot.core.uniprotkb.taxonomy.impl.TaxonomyBuilder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class TaxonomyDocumentConverterTest {
+
+    private static ObjectMapper mapper = TaxonomyJsonConfig.getInstance().getFullObjectMapper();
+
+    @Test
+    void convertMinimal() throws IOException {
+        TaxonomyEntry entry =
+                new TaxonomyEntryBuilder()
+                        .taxonId(10L)
+                        .parent(new TaxonomyBuilder().taxonId(1L).build())
+                        .rank(TaxonomyRank.NO_RANK)
+                        .build();
+        TaxonomyDocumentConverter converter = new TaxonomyDocumentConverter(mapper);
+        TaxonomyDocument result = converter.convert(entry);
+        assertNotNull(result);
+        assertEquals(10L, result.getTaxId());
+        assertEquals(TaxonomyRank.NO_RANK.getName(), result.getRank());
+        assertNull(result.getParent());
+
+        assertNotNull(result.getTaxonomyObj());
+        TaxonomyEntry documentEntry =
+                mapper.readValue(result.getTaxonomyObj(), TaxonomyEntry.class);
+        assertEquals(entry.getTaxonId(), documentEntry.getTaxonId());
+        assertEquals(entry.getRank(), documentEntry.getRank());
+        assertNull(documentEntry.getParent());
+    }
+
+    @Test
+    void convertFullEntry() throws IOException {
+        TaxonomyEntry entry = getEntry();
+        TaxonomyDocumentConverter converter = new TaxonomyDocumentConverter(mapper);
+        TaxonomyDocument result = converter.convert(entry);
+        assertNotNull(result);
```

Comment:
optional - if we rename "result" variable to taxonomyDocument or similar, it would be easier to read. But it is nice to have.